registry/auth/token/accesscontroller.go

```diff
@@ -102,13 +105,36 @@ func (ac authChallenge) Status() int {
 	return http.StatusUnauthorized
 }
 
+func buildAutoRedirectURL(r *http.Request, autoRedirectPath string, forceTLS bool) string {
+	scheme := "https"
```

Comment:
Did you mean to set this to "plain" `http`?

Reply:
`https` as default.